dotnet/src/Connectors/Connectors.HuggingFace/TextGeneration/TextGenerationStreamJsonParser.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json.Nodes;
+using Microsoft.SemanticKernel.Connectors.HuggingFace.Core;
+
+namespace Microsoft.SemanticKernel.Connectors.HuggingFace.TextGeneration;
+
+internal sealed class TextGenerationStreamJsonParser : IStreamJsonParser
```

Comment:
Can we do something like this if we are validating as json and otherwise just read as a string?

```
var serializer = new JsonSerializer();
using (var sr = new StreamReader(responseMessage))
using (var jsonTextReader = new JsonTextReader(sr))
{
    var jsObj= serializer.Deserialize(jsonTextReader);
}
```

What is the unit test coverage for this class? If we need this we want to ensure it is high and also we have negative test cases e.g. badly formed json with validation turned on.

Reply:
Added the unit tests for this class, used the same tests from Gemini, quite useful class BTW, it reads the stream and split the different json blocks as they arrive {} {} {}